VocaDbWeb.Core/Models/Album/AlbumEditViewModel.cs

```diff
@@ -0,0 +1,90 @@
+#nullable disable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using VocaDb.Model;
+using VocaDb.Model.DataContracts.Albums;
+using VocaDb.Model.DataContracts.UseCases;
+using VocaDb.Model.Domain;
+using VocaDb.Model.Domain.Albums;
+using VocaDb.Model.Domain.Security;
+using VocaDb.Model.Utils;
+using VocaDb.Web.Code;
+using VocaDb.Web.Code.Exceptions;
+using VocaDb.Web.Helpers;
+
+namespace VocaDb.Web.Models.Album
+{
+	public class AlbumEditViewModel
+	{
+		public AlbumEditViewModel()
+		{
+			AllDiscTypes = Translate.DiscTypeNames.GetValuesAndNamesStrings(AppConfig.AlbumTypes);
+
+			DiscTypeDescriptions = ViewRes.Album.EditStrings.BaDiscTypeExplanation
+								   + "<br /><br /><ul>" + string.Join("",
+									   EnumVal<DiscType>.Values.Where(v => v != DiscType.Unknown).Select(v => $"<li><strong>{Translate.DiscTypeName(v)}</strong>: {(global::Resources.DiscTypeDescriptions.ResourceManager.GetString(v.ToString()))}</li>"));
+		}
+
+		public AlbumEditViewModel(AlbumContract album, IUserPermissionContext permissionContext,
+			bool canDelete,
+			AlbumForEditContract editedAlbum = null)
+			: this()
+		{
+			ParamIs.NotNull(() => album);
+
+			Album = album;
+			CanDelete = canDelete;
+			EditedAlbum = editedAlbum;
+
+			AllowedEntryStatuses = EntryPermissionManager.AllowedEntryStatuses(permissionContext).ToArray();
+		}
+
+		public AlbumContract Album { get; set; }
+
+		public Dictionary<string, string> AllDiscTypes { get; set; }
+
+		public EntryStatus[] AllowedEntryStatuses { get; set; }
+
+		public bool CanDelete { get; set; }
+
+		public string DiscTypeDescriptions { get; set; }
+
+		public bool Draft => Album != null && Album.Status == EntryStatus.Draft;
+
+		[ModelBinder(BinderType = typeof(JsonModelBinder))]
+		// TODO: implement [AllowHtml]

```

Comment:
Why is the `AllowHtml` attribute needed here?

Reply:
Album description at least may contain characters that are interpreted as HTML. Otherwise ASP.NET request validation would block all ```<``` and ```>``` at least.